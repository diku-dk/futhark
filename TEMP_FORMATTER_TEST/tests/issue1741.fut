-- ==
-- error: type Op.t = i64

type L0 = i64

type L1 = [1]i64


module type Const = {
    type t
    val x: t
}

module L0: Const = {
    type t = L0
    def x : t = 0
}


module L1: Const = {
    type t = L1
    def x : t = [1i64]
}

module type Op = (X: Const) -> Const

module type Container = {
    module Op: Op
}

module L0Op: Container = {
    module Op (X: Const with t = L0): Const = {
        type t = L0
        def x = X.x + 1
    }
}


module L2 = L0Op.Op(L1)

entry main = L2.x
