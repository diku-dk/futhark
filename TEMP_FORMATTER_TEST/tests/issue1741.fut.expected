8dff4a06d57712bbd988d4a84ec0e7a3
